src/bokeh/models/nodes.py

```diff
@@ -0,0 +1,214 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2023, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Standard library imports
+from typing import Any, Literal
+
+# Bokeh imports
+from ..core.properties import (
+    Either,
+    Enum,
+    Instance,
+    Int,
+    Required,
+    String,
+)
+from ..core.property.aliases import CoordinateLike
+from ..model import Model
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    "Indexed",
+    "Node",
+    "XY",
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+ImplicitTarget = Literal["viewport", "canvas", "plot", "frame", "parent"]
+
+class Coordinate(Model):
+    """
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+class XY(Coordinate):
+    """
+    A point in a Cartesian coordinate system.
+
+    .. note::
+        This model is experimental and may change at any point.
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    x = Required(CoordinateLike, help="""
+    The x component.
+    """)
+
+    y = Required(CoordinateLike, help="""
+    The y component.
+    """)
+
+class Indexed(Coordinate):
+    """
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    i = Required(Int())
+    renderer = Instance(".models.renderers.GlyphRenderer")
```

Comment:
```suggestion
    i = Required(Int())
    
    renderer = Instance(".models.renderers.GlyphRenderer")
```
minimal help strings would be nice as well